providers/internal_util.go

```diff
@@ -9,6 +9,20 @@ import (
 	"github.com/pusher/oauth2_proxy/pkg/requests"
 )
 
+type NotImplementedError struct {
+	message string
+}
+
+func NewNotImplementedError(message string) *NotImplementedError {
```

Comment:
What's the benefit of adding this, I like the style but I can only see it being used once at present, are we comparing it somewhere?